### ISSUE TITLE
[ENH] `MSTL` transformer should allow to return individual seasonality components as `transform` results

### ISSUE THREAD

**Issue by fkiraly** [sktime/sktime#5458]
It should be possible for the `MSTL` transformer to return individual seasonality components as `transform` results - then it can be used with individual forecasters on the components, e.g., naive seasonal forecasters on the seasonal components.

For this, we need to take into account:

* some convention on the naming of the return components. I would simply name them `seasonal__7` and so on, with double-underscore which is the `sktime` convention for nested fields, and with `7` being the string coercion of periodicity.
* the option should be enabled by a new parameter imo, or a modification to the `return_components` parameter. In order to not break downstream code, the default should remain as current.

Related to the discussion in https://github.com/sktime/sktime/issues/5454, FYI @arturdragunov (also in case you may like to pick this up)

**Comment by fkiraly** [comment]
the point is, the `transform` only returns a `"transformed"` column, not the `"seasonal_3"` columns etc.

This should be possible, and switched on via the `return_components` argument

**Comment by Abhay-Lejith** [comment]
Just checked this out. It seems to be returning the seasonal components too.
```python
from sktime.datasets import load_airline
from sktime.transformations.series.detrend.mstl import MSTL

y = load_airline()
y.index = y.index.to_timestamp()
mstl = MSTL(periods=[3,12],return_components=True)  
mstl.fit(y)  
ret=mstl.transform(y)
print(ret)
```
```
            transformed       trend     resid  seasonal_3  seasonal_12
Period                                                                
1949-01-01   118.908461  120.868660 -1.960199    1.309070    -8.217532
1949-02-01   117.366632  121.884784 -4.518152   -3.531810     4.165178
1949-03-01   119.036115  122.856305 -3.820190    2.487008    10.476878
1949-04-01   122.672055  123.785125 -1.113070    1.033097     5.294849
1949-05-01   124.537851  124.678598 -0.140746   -3.775078     0.237226
...                 ...         ...       ...         ...          ...
1960-08-01   476.819239  481.971978 -5.152738  -13.866788   143.047549
1960-09-01   486.496353  485.584534  0.911819   -1.848749    23.352396
1960-10-01   488.452586  489.179596 -0.727009   17.157415   -44.610001
1960-11-01   490.664258  492.757427 -2.093169  -15.707845   -84.956413
1960-12-01   496.570297  496.319199  0.251098   -2.505060   -62.065237
```

**Comment by fkiraly** [comment]
oh yes, it does!

Maybe this is outdated then?

In either case, it would be nice to add a test that checks whether columns with the expected components are indeed returned.

**Comment by kcentric** [comment]
@fkiraly Working on the test right now. Yes, strange that we have tests for the other deseasonalize and detrend but not for mstl. 

Forgot to comment but just doing that so someone doesn't pick up this issue before I'm done.

**Comment by fkiraly** [comment]
Let's close this issue then together with #6083 once #6084 is merged.

We can link both issues by writing closing keywords like "closes #6083" in the PR description. It needs to be one of a couple, "addresses" is not among them.

**Comment by kcentric** [comment]
Oh I see. Yes, I just said "addresses" because I thought my current test case isn't testing _all_ functionalities of MSTL, so you might've wanted me to write more test cases before closing the issue. But then, we could do all that expansion within the PR itself before merging it, so yes, makes sense!
